src/Jobs/UpdateSyncableProductsCount.php

```diff
@@ -0,0 +1,115 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Jobs;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\ActionScheduler\ActionSchedulerInterface;
+use Automattic\WooCommerce\GoogleListingsAndAds\Jobs\AbstractBatchedActionSchedulerJob;
+use Automattic\WooCommerce\GoogleListingsAndAds\Jobs\ActionSchedulerJobMonitor;
+use Automattic\WooCommerce\GoogleListingsAndAds\Jobs\SyncableProductsBatchedActionSchedulerJobTrait;
+use Automattic\WooCommerce\GoogleListingsAndAds\Options\TransientsInterface;
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\ProductRepository;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class UpdateSyncableProductsCount
+ *
+ * Get the number of syncable products (i.e. product ready to be synced to Google Merchant Center) and update it in the DB.
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Jobs
+ * @since x.x.x
+ */
+class UpdateSyncableProductsCount extends AbstractBatchedActionSchedulerJob {
+	use SyncableProductsBatchedActionSchedulerJobTrait;
+
+	/**
+	 * @var ProductRepository
+	 */
+	protected $product_repository;
+
+	/**
+	 * @var TransientsInterface
+	 */
+	protected $transients;
+
+	/**
+	 * @var int
+	 */
+	private $count;
+
+	/**
+	 * UpdateSyncableProductsCount constructor.
+	 *
+	 * @param ActionSchedulerInterface  $action_scheduler
+	 * @param ActionSchedulerJobMonitor $monitor
+	 * @param ProductRepository         $product_repository
+	 * @param TransientsInterface       $transients
+	 */
+	public function __construct( ActionSchedulerInterface $action_scheduler, ActionSchedulerJobMonitor $monitor, ProductRepository $product_repository, TransientsInterface $transients ) {
+		parent::__construct( $action_scheduler, $monitor );
+		$this->product_repository = $product_repository;
+		$this->transients         = $transients;
+		$this->count              = 0;
+	}
+
+	/**
+	 * Get the count instance variable.
+	 *
+	 * @return int
+	 */
+	public function get_count(): int {
+		return $this->count;
+	}
+
+	/**
+	 * Set the count instance variable.
+	 *
+	 * @param int $count The number that will be set to $this->count.
+	 */
+	public function set_count( $count ) {
+		$this->count = $count;
```

Comment:
This seems to only work because several batches are handled quickly in one request. But if the batches are spread out then it should have a new instance of the class so it would no longer persist the counted data.

We need to somehow persist the in between data, which I'm assuming needs to be a list of product ID's to properly aggregate product variations (or is that planned for a followup PR?)

Reply:
> This seems to only work because several batches are handled quickly in one request. But if the batches are spread out then it should have a new instance of the class so it would no longer persist the counted data.

Thanks, I was also thinking about it but I did not run into the problem so I assumed async scheduler would always use the same process for the same batched job. Obviously I was wrong so I will update it to keep the intermediate data in the transient.

> which I'm assuming needs to be a list of product ID's to properly aggregate product variations (or is that planned for a followup PR?)

Yeah, aggregating the product variations will be in a followup PR. In this PR I'm going to store a list of product IDs in a transient and append the list in each batch until the job is done.

Reply:
> In this PR I'm going to store a list of product IDs in a transient

Update: will use options instead of transients, since the spirit of transients is, [it might disappear one second after you set them, or 24 hours, but they will never be around after the expiration time](https://developer.wordpress.org/apis/handbook/transients/).

Reply:
Updated in aaf27ae.